internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+		return
+	}
+	waitParams := compute.Wait{
+		ClusterID: installLib.ClusterId,
+		IsRunning: true,
+	}
+	libraryRep := libGoSDK.String()
+	installedLib := LibraryExtended{}
+
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &installedLib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, installedLib)...)
+}
+
+func (r *LibraryResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libraryTfSDK LibraryExtended
+	resp.Diagnostics.Append(req.State.Get(ctx, &libraryTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libGoSDK compute.Library
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, libraryTfSDK, &libGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clusterId := libraryTfSDK.ClusterId.ValueString()
+	libraryRep := libGoSDK.String()
+	installedLib := LibraryExtended{}
+	waitParams := compute.Wait{
+		ClusterID: clusterId,
+		IsRefresh: true,
+	}
+
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &installedLib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, installedLib)...)
+}
+
+func (r *LibraryResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+}
+
+func (r *LibraryResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libraryTfSDK LibraryExtended
+	resp.Diagnostics.Append(req.State.Get(ctx, &libraryTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clusterID := libraryTfSDK.ClusterId.ValueString()
+	var libGoSDK compute.Library
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, libraryTfSDK, &libGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	libraryRep := libGoSDK.String()
+	_, err := clusters.StartClusterAndGetInfo(ctx, w, clusterID)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to start and get cluster", err.Error())
+		return
+	}
+	cll, err := w.Libraries.ClusterStatusByClusterId(ctx, clusterID)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to get libraries", err.Error())
+		return
+	}
+	for _, v := range cll.LibraryStatuses {
+		if v.Library.String() != libraryRep {
+			continue
+		}
+		err := w.Libraries.Uninstall(ctx, compute.UninstallLibraries{
+			ClusterId: clusterID,
+			Libraries: []compute.Library{*v.Library},
+		})
+		if err != nil {
+			resp.Diagnostics.AddError("failed to uninstall library", err.Error())
+		}
+		return
+	}
+	resp.Diagnostics.AddError("failed to uninstall library", fmt.Sprintf("failed to find %s on %s", libraryRep, clusterID))
```

Comment:
I see that you're porting the existing implementation, but this doesn't seem like an error to me. E.g. if the library was manually uninstalled. Can you leave a comment here about this?